src/ring_theory/witt_vector/init_tail.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2020 Johan Commelin and Robert Y. Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Robert Y. Lewis
+-/
+
+import ring_theory.witt_vector.basic
+import ring_theory.witt_vector.is_poly
+
+/-!
+
+# `init` and `tail`
+
+Given a Witt vector `x`, we are sometimes interested
+in its components before and after an index `n`.
+This file defines those operations, proves that `init` is polynomial,
+and shows how that polynomial interacts with `mv_polynomial.bind₁`.
+
+## Main declarations
+
+* `witt_vector.init n x`: the first `n` coefficients of `x`, as a Witt vector. All coefficients at
+  indices ≥ `n` are 0.
+* `witt_vector.tail n x`: the complementary part to `init`. All coefficients at indices < `n` are 0,
+  otherwise they are the same as in `x`.
+* `witt_vector.coeff_add_of_disjoint`: if `x` and `y` are Witt vectors such that for every `n`
+  the `n`-th coefficient of `x` or of `y` is `0`, then the coefficients of `x + y`
+  are just `x.coeff n + y.coeff n`.
+-/
+
+variables {p : ℕ} [hp : fact p.prime] (n : ℕ) {R : Type*} [comm_ring R]
+
+local notation `𝕎` := witt_vector p -- type as `\bbW`
+
+namespace tactic
+namespace interactive
+setup_tactic_parser
+
+/--
+`init_ring` is an auxiliary tactic that discharges goals factoring `init` over ring operations.
+-/
+meta def init_ring (assert : parse (tk "using" >> parser.pexpr)?) : tactic unit := do
+`[rw ext_iff,
+  intros i,
+  simp only [init, select, coeff_mk],
+  split_ifs with hi; try {refl}],
+match assert with
+| none := skip
+| some e := do
+  `[simp only [add_coeff, mul_coeff, neg_coeff],
+    apply eval₂_hom_congr' (ring_hom.ext_int _ _) _ rfl,
+    rintro ⟨b, k⟩ h -],
+  tactic.replace `h ```(%%e p _ h),
+  `[simp only [finset.mem_range, finset.mem_product, true_and, finset.mem_univ] at h,
+    have hk : k < n, by linarith,
+    fin_cases b;
+    simp only [function.uncurry, matrix.cons_val_zero, matrix.head_cons, coeff_mk,
+      matrix.cons_val_one, coeff_mk, hk, if_true]]
+end
+
+end interactive
+end tactic
+
+namespace witt_vector
+open mv_polynomial
+
+open_locale classical
+noncomputable theory
+
+section
+
+local attribute [semireducible] witt_vector
+
+/-- `witt_vector.select P x`, for a (decidable) predicate `P : ℕ → Prop` is the Witt vector
```

Comment:
Since we don't have a `decidable` predicate, should we leave this out?
```suggestion
/-- `witt_vector.select P x`, for a predicate `P : ℕ → Prop` is the Witt vector
```